lib/optimizely/event_builder.rb

```diff
@@ -6,28 +6,18 @@ module Optimizely
   class Event
     # Representation of an event which can be sent to the Optimizely logging endpoint.
 
-    # Event API format
-    OFFLINE_API_PATH = 'https://%{project_id}.log.optimizely.com/event'
-
-    # Gets/Sets event params.
-    attr_accessor :params
+    attr_reader :method
+    attr_reader :params
+    attr_reader :url
 
-    def initialize(params)
+    def initialize(method, url, params)
+      @method = method
+      @url = url
       @params = params
     end
-
-    def url
-      # URL for sending impression/conversion event.
-      #
-      # project_id - ID for the project.
-      #
-      # Returns URL for event API.
-
-      sprintf(OFFLINE_API_PATH, project_id: @params[Params::PROJECT_ID])
-    end
   end
 
-  class EventBuilder
+  class EventBuilderV1
```

Comment:
How about this, since the `EventBuilder` is already initialized with the project config (which tells us whether we are in V1 or V2), why don't we just have the event builder determine whether to return a V1 Event or V2 event when we call eventBuilder.getConversionEvent and eventBuilder.getImpressionEvent. That event can be passed into event dispatcher, which determines based on the event whether to get or post